src/App.tsx

```diff
@@ -1,73 +1,133 @@
-import React from 'react';
+import React, { useEffect, useMemo, useState } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import classNames from 'classnames';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+type Props = {
+  debounceDelay?: number;
+  onSelected?: (person: Person | null) => void;
+};
+
+export const App: React.FC<Props> = ({ debounceDelay = 300, onSelected }) => {
+  const [query, setQuery] = useState('');
+  const [error, setError] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [isDropdownActive, setIsDropdownActive] = useState(false);
+  const [debouncedQuery, setDebouncedQuery] = useState(query);
+
+  useEffect(() => {
+    const handler = setTimeout(() => {
+      setDebouncedQuery(query);
+    }, debounceDelay);
+
+    return () => {
+      clearTimeout(handler);
+    };
+  }, [query, debounceDelay]);
+
+  const filteredPeople = useMemo(() => {
+    if (debouncedQuery === '') {
+      return peopleFromServer;
+    }
+
+    const filteredArr = peopleFromServer.filter(person =>
+      person.name.toLowerCase().includes(debouncedQuery.toLowerCase()),
+    );
+
+    setError(filteredArr.length === 0);
```

Comment:
Setting the error state inside the `useMemo` hook might lead to unnecessary state updates. Consider moving this logic outside of the `useMemo` to ensure that the error state is only updated when necessary.